### PR TITLE
Issues fixes #6493 and #6485

### LINKED ISSUE #6493: App crashes when we edit the route and press X button of a point
Using Osmand~ 3.3.0 26668 on Asus ZenFone 5z running on Android 8.0.0.

When I press **Edit** button in the Route editor, and press on the X button of any point, the app crashes. I have already sent three crash reports.

But if I swipe the point, I can remove the point without crashing the app.

### PR DESCRIPTION
**Pull request by MadWasp79**
a couple of small issues:
https://github.com/osmandapp/Osmand/issues/6493
https://github.com/osmandapp/Osmand/issues/6485

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/routepreparationmenu/WaypointsFragment.java [hunk 1]
```diff
@@ -587,7 +587,7 @@ public void onItemClick(AdapterView<?> parent, View view, int position, long id)
 	}
 
 	private void deleteSwipeItem(int position) {
-		if (swipeDismissListener != null) {
+		if (swipeDismissListener != null && position < listView.getCount()) {
```
Comment (crimean) [inline OsmAnd/src/net/osmand/plus/routepreparationmenu/WaypointsFragment.java]: looks like wrong fix

Comment (crimean) [reply OsmAnd/src/net/osmand/plus/routepreparationmenu/WaypointsFragment.java]: reason of the issue:
https://github.com/osmandapp/Osmand/commit/bdf3ad58ef1ac6d5cf1f71014564d1d8d7953f37#diff-234d288dc83cf5d5e8ca2711f14e9b28R94

File: OsmAnd/src/net/osmand/plus/rastermaps/OsmandRasterMapsPlugin.java [hunk 2]
```diff
@@ -286,6 +285,9 @@ public void onMapSelected(boolean canceled) {
 										adapter.notifyDataSetChanged();
 									}
 								});
+						if (!isChecked) {
+							hideSeekBar(mapActivity, mapActivity.getMyApplication().getSettings().MAP_OVERLAY_TRANSPARENCY);
```
Comment (crimean) [inline OsmAnd/src/net/osmand/plus/rastermaps/OsmandRasterMapsPlugin.java]: this should be implemented in toggleUnderlayState

File: OsmAnd/src/net/osmand/plus/rastermaps/OsmandRasterMapsPlugin.java [hunk 3]
```diff
@@ -582,10 +587,9 @@ public void toggleUnderlayState(@NonNull MapActivity mapActivity,
...
 		}
-
-		boolean isChecked = map == null;
+		boolean isChecked = map != null;
 		boolean showSeekbar = isChecked && RasterMapMenu.isSeekbarVisible(app, type);
-		boolean hideSeekbar = !isChecked && RasterMapMenu.isSeekbarVisible(app, type);
+		boolean hideSeekbar = !isChecked;
 		MapActivityLayers mapLayers = mapActivity.getMapLayers();
 		CommonPreference<LayerTransparencySeekbarMode> seekbarModePref = settings.LAYER_TRANSPARENCY_SEEKBAR_MODE;
 		if (showSeekbar) {
```
Comment (crimean) [inline OsmAnd/src/net/osmand/plus/rastermaps/OsmandRasterMapsPlugin.java]: Did you trace showTransparencyBar, hideTransparencyBar, setTransparencyBarEnabled methods across the code? There might be more bugs related to show/hide the bar.

File: OsmAnd/src/net/osmand/plus/routepreparationmenu/WaypointsFragment.java (hunk 1 above)
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/plus/routepreparationmenu/WaypointsFragment.java]: ok, removed.

**Pull request merged**